bot/admin/test/xray/src/main/kotlin/XrayService.kt

```diff
@@ -582,15 +584,75 @@ class XrayService(
             testsPlans: List<String>,
             labelTestPlansMap: Map<String, String> = emptyMap()
     ): XrayTest? {
+        val steps = defineTestSteps(dialog)
+
+        val labels = linkedJira?.let { XrayClient.getLabels(it) } ?: emptyList()
+
+        val test = JiraTest(
+                jiraProject,
+                testName.invoke(labels),
+                "",
+                listOf("automatisation"),
+                testTypeField,
+                manualStepsField,
+                "cucumber-tock"
+        )
+        val jira = XrayClient.createTest(test)
+        if (linkedJira != null) {
+            XrayClient.linkTest(jira.key, linkedJira)
+            testsPlans.forEach {
+                XrayClient.addTestToTestPlan(jira.key, it)
+            }
+            if (labelTestPlansMap.isNotEmpty()) {
+                labels
+                        .filter { labelTestPlansMap.containsKey(it) }
+                        .forEach {
+                            XrayClient.addTestToTestPlan(jira.key, labelTestPlansMap[it]!!)
+                        }
+            }
+        }
+
+        steps?.forEach {
+            XrayClient.saveStep(jira.key, it)
+        }
+
+        XrayPrecondition.getPreconditionForUserInterface(dialog.actions.first().userInterfaceType)
```

Comment:
Are we sure that `actions` will never be empty at this stage  ? If not `first()` failure should be handled

Reply:
I don't know much about XrayPrecondition. I just extracted this step definition from generateXray() because this part would be used bot in generateXray() and updateXrayTest() methods.
I think that dialog would be filled at least with a greeting but I'm not sure. @vsct-jburet may answer better than me to this :).

Reply:
After discussing with @vsct-jburet , dialog contains at least a greeting, then dialog is never empty.